### ISSUE TITLE
Build model comparison and hyperparameter tuning framework

### ISSUE THREAD

**Issue by muk2** [Kame4201/beat-books-model#24]
### Why this matters
**Non-technical:** We have multiple model types (Logistic Regression, XGBoost, LightGBM) but no systematic way to compare them. Which model actually makes the best predictions? Which settings work best? Without a comparison framework, we're guessing. This is like having multiple employees but no way to evaluate their performance — you need metrics and fair tests to know who's best at what.

**Technical:** Model selection and hyperparameter tuning are critical for maximizing prediction quality. The walk-forward validation framework (PR #7) provides the backtest infrastructure, but we need an orchestration layer that runs multiple models through the same backtests and ranks them.

### Implementation Plan

1. **Model comparison runner** (`src/models/comparison.py`):
   ```python
   class ModelComparison:
       def compare(self, models: List[Model], data, features) -> ComparisonResult:
           """Run each model through identical walk-forward backtests."""
           results = {}
           for model in models:
               bt = Backtester(config)
               results[model.name] = bt.run(data, model, features)
           return self.rank_models(results)
   ```

2. **Hyperparameter tuning** (`src/models/tuning.py`):
   - Grid search or Bayesian optimization within walk-forward framework
   - CRITICAL: Tuning must use walk-forward CV, NOT random K-fold (per CLAUDE.md rules)
   - Inner loop: tune hyperparams on expanding window
   - Outer loop: evaluate tuned model on held-out future period

3. **Ranking and selection**:
   - Primary metric: ROI (what makes money)
   - Secondary metrics: Accuracy, Brier score, Sharpe ratio
   - Confidence intervals via bootstrap
   - Visualization of model performance comparison

4. **Model registry integration**:
   - Best model auto-promoted to "production" in registry
   - Track which model version is currently "live"
   - A/B testing between model versions

### Output
```
Model Comparison Report (2022-2024 backtest):
┌──────────────┬──────────┬────────┬────────┬─────────┐
│ Model        │ Accuracy │ ROI    │ Sharpe │ Brier   │
├──────────────┼──────────┼────────┼────────┼─────────┤
│ XGBoost v2   │ 58.3%    │ +4.2%  │ 0.31   │ 0.238   │
│ LightGBM v1  │ 57.1%    │ +2.8%  │ 0.22   │ 0.241   │
│ LogReg v3    │ 55.9%    │ +0.5%  │ 0.08   │ 0.245   │
└──────────────┴──────────┴────────┴────────┴─────────┘
Winner: XGBoost v2 (highest ROI + Sharpe)
```

### Acceptance Criteria
- [ ] `ModelComparison` class runs N models through same backtest
- [ ] Hyperparameter tuning with walk-forward CV (no leakage)
- [ ] Ranking table with multiple metrics
- [ ] Confidence intervals for key metrics
- [ ] Integration with model registry for promotion
- [ ] CLI: `python -m src.models.compare --models xgb,lgb,lr --seasons 2020-2024`

### Dependencies
- Requires PRs #5, #6, #7 merged
- Requires #19 (pipeline CLI) for orchestration

**Comment by Kame4201** [comment]
**Triage: KEEP — partially implemented.**

**What exists:** `ModelRegistry` (version tracking), `model_config.py` (hyperparameter configs for 4 Win/Loss + 4 Spread variants)
**What's missing:** Automated hyperparameter search, cross-model comparison framework, experiment tracking

**Acceptance criteria:**
- [ ] `src/models/tuner.py` — hyperparameter search (GridSearch / Optuna)
- [ ] `src/models/comparator.py` — run N models on same data, rank by metrics
- [ ] Experiment logging (MLflow or lightweight JSON-based)
- [ ] CLI integration: `python -m src.models --tune --model win_loss`
- [ ] Comparison report output (table of model × metric)
- [ ] Unit tests for tuner + comparator

**Implementation hints:**
- Build on existing `model_config.py` param grids
- `ModelRegistry` already tracks metadata — extend with experiment results
- Tests: `tests/test_models/test_tuner.py`, `test_comparator.py`

**Dependencies:** None (uses existing model infrastructure)
**Branch:** `feature/issue-24-model-comparison` (created)
